snuba/datasets/querylog.py

```diff
@@ -0,0 +1,94 @@
+from datetime import timedelta
+from snuba import settings
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    Enum,
+    Float,
+    LowCardinality,
+    Mapping,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+)
+
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.datasets.querylog_processor import QuerylogProcessor
+from snuba.query.extensions import QueryExtension
+from snuba.query.project_extension import ProjectExtension, ProjectExtensionProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class QuerylogDataset(TimeSeriesDataset):
+    def __init__(self) -> None:
+        status_type = Enum([("success", 0), ("error", 1), ("rate-limited", 2)])
+
+        columns = ColumnSet(
+            [
+                ("request", String()),
+                ("referrer", LowCardinality(String())),
+                ("dataset", LowCardinality(String())),
+                ("projects", Array(UInt(64))),
+                ("organization", Nullable(UInt(64))),
+                ("timestamp", DateTime()),
+                ("duration_ms", UInt(32)),
+                ("status", status_type),
+                (
+                    "clickhouse_queries",
+                    Nested(
+                        [
+                            ("sql", String()),
+                            ("status", status_type),
+                            ("final", UInt(8)),
+                            ("cache_hit", UInt(8)),
+                            ("sample", Float(32)),
+                            ("max_threads", UInt(8)),
+                            ("duration_ms", UInt(32)),
+                            ("trace_id", Nullable(UUID())),
```

Comment:
Why not adding the full stats field ? There are a few other interesting fields inside like whether the query is consistent, the results size, data about the rate limiting, etc.

Reply:
ping

Reply:
Are you suggesting adding some of those fields at a top level or having stats as a nested field? Personally I find it easier to write queries quickly without ARRAY JOIN (especially if we are already doing dealing with other nested fields in the dataset)

Reply:
After talking in person, in order to keep the dataset schema stable, it is better adding first the columns we need and not the stats payload as it is.